rmw_connextdds_common/src/ndds/dds_api_ndds.cpp

```diff
@@ -207,6 +207,12 @@ rmw_connextdds_initialize_participant_qos_impl(
   }
 #endif /* RMW_CONNEXT_FAST_ENDPOINT_DISCOVERY */
 
+#if RMW_CONNEXT_SHARE_DDS_ENTITIES_WITH_CPP
+  dp_qos->user_object.flow_controller_user_object.size = sizeof(void*);
+  dp_qos->user_object.topic_user_object.size = sizeof(void*);
+  dp_qos->user_object.content_filtered_topic_user_object.size = sizeof(void*);
```

Comment:
I don't quite understand this, but if this allows sharing a participant with a C++ application I'm happy with it.

Reply:
Yeah, I don't quite understand it either, but I've verified internally at RTI that it doesn't affect the C layer.

Reply:
Hmm, it'd be good to leave at least a hint as to what is this doing for our future selves sanity.

Reply:
For now my assumption was that the guard with `RMW_CONNEXT_SHARE_DDS_ENTITIES_WITH_CPP` and [the associated comment in `static_config.h`](https://github.com/ros2/rmw_connextdds/pull/25/files/a9bf187189544db74aab71d756225153c6834edb#diff-946f3d8da82c81b526adee816615596f0325e8416167d674b2fbdfc9db7d7761R283) would be enough to keep track of the nature of this code, but I can add another comment here too if preferred.

Reply:
Those are fine. A hint as to why these three lines have that effect is what I had in mind. Something like a reference to the corresponding RTI Connext documentation would be enough.

Reply:
This is an internal, undocumented QosPolicy so there isn't anything to link. It is only used by the (implementations o) various Connext language bindings which are built on top of the C libraries.

AFAICT, in this case it is instructing some dependant logic to treat some objects (flow controllers, topics, and cft's) as having size in memory equal to a pointer. The requirement is dictated by the implementation of the Modern C++ layer. Other language bindings might set these values (and other in the QosPolicy which are not set in this case) to some other values, depending on their implementations. The only thing more I can tell about these settings is that if you don't customize the QoS, and try to create a Topic, you'll end up with a SIGSEGV :)

Maybe I can add a comment about them being internal, undocumented settings required by the Connext C++11 API.

Reply:
[Added a comment](https://github.com/ros2/rmw_connextdds/pull/25/files#diff-15424b0249cc41e5d4179bade17b540ec7d1edd8b20f7d60bc614fe7be0316d3R211)